COMPILING.md

```diff
@@ -371,28 +371,36 @@ compilation flag:
 ## Compiling with alternative SAT solvers
 
 For the packaged builds of CBMC on our release page we currently build CBMC
-with the MiniSat2 SAT solver statically linked at compile time. However it is
-also possible to build CBMC using alternative SAT solvers.
+with the MiniSat2 and CaDiCaL SAT solvers statically linked at compile time.
+However it is also possible to build CBMC using alternative SAT solvers.
 
 ### Compiling CBMC Using Solver Native Interfaces
 
 The following solvers are supported by CBMC using custom interfaces and can
-by downloaded and compiled by the build process: MiniSAT2, CaDiCaL, and Glucose.
+be downloaded and compiled by the build process: MiniSAT2, CaDiCaL, and Glucose.
 
-For `make` alternatives to the default (i.e. not MiniSAT) can be built with the
-following commands for CaDiCaL:
-```
-make -C src cadical-download
-make -C src CADICAL=../../cadical
-```
-and for glucose
+For `make`, alternatives to the default (i.e. not MiniSAT and CaDiCaL) can be
+built with the following commands for glucose:
 ```
 make -C src glucose-download
 make -C src GLUCOSE=../../glucose-syrup
 ```
+CBMC can be built with multiple solvers, which can then be selected at runtime
+using the `--sat-solver` option.
+For example, to build CBMC with MiniSAT2 and Glucose, do:
+```
+make -C src minisat2-download glucose-download
+make -C src MINISAT2=../../minisat-2.2.1 GLUCOSE=../../glucose-syrup
+```
+In that case, MiniSAT2 will be used by default. To switch to glucose at runtime,
+run CBMC with `--sat-solver glucose`.
```

Comment:
This implies that MiniSAT2 is the default because it appears first in the variable arguments to the `make` command. I think we should be clear on what the default (or default ordering) is after this PR is merged.

Reply:
The ordering is actually specified by the order in this `#if` block:

https://github.com/diffblue/cbmc/blob/develop/src/solvers/sat/satcheck.h#L62

The first one that is included in the build becomes the default solver.

This was the case even before this PR: if CBMC is built with `make -C src GLUCOSE=../../glucose-syrup LINGELING=../../lingeling ` for example, lingeling will be used because it appears first in this `#if` block.

Should we mention this order in `COMPILING.md`? If so, which subset of solvers should we mention? Currently, only minisat2, cadical, and glucose are mentioned in `COMPILING.md`.

Reply:
Given the open source nature of CBMC I would propose that the ordering is either:

1. Listed in the documentation explicitly (but this may go out of date if the code changes); or
2. defined in the documentation according to the code with a link to the appropriate code (which requires code basic reading ability, but is resistant to code changes).

Reply:
I expanded the documentation and pointed to the code that decides the default solver.